### ISSUE TITLE
CF: Update display of Caution Flags on the Profile Page

### ISSUE THREAD

**Issue by desireeturner** [department-of-veterans-affairs/va.gov-team#6805]
As a Veteran searching for institutions using the GI Bill Comparison Tool, I need to know what types of caution flags the institutions I'm considering have so that I can make the best decision about my education choices.

## Assumptions: 
1. There will be a follow on story (#7151) to map the content included in supporting artifacts to what is being displayed in the UI. To test this story, mock data will be used. 
2. There will be a follow on story to this to update final approved content. 
3. This story should be reviewed by a member of the UX team prior to completion for feedback.
4. This story will be behind a prod flag so that the content is only available on staging when merged.

## Acceptance Criteria: 
1. If a school has closing flags, the profile page includes all applicable warnings at the top of the profile page. 
a. School closings include a header that states "School Closing" 
b. Below the header is messaging as well as links to the school's website for more information if a school website exists
c. The school closing is displayed separately above any caution flags for the institution

2. If a school has caution flags, the profile page includes all applicable warnings at the top of the profile page. 
a. Caution flags that are NOT school closings included a header that states: "This school has cautionary warnings" or “This school has a cautionary warning”. 
b. Below the header specific subtext regarding the type of caution flag(s) is included 
c. An anchor link to the cautionary information section of the profile page is included with the caution flags. 

3. In the cautionary information section of the institution profile page: 
a. an alert exists for each of the institution's caution flags that includes a header, subtext and in some cases a link and linked text to supporting information.
b. Below the alerts is informational text related to cautionary warnings in general as well as a link to the "About this Tool" page. 
c. A "Student Feedback" section header exists above the student feedback section. 

Supporting Artifacts: 
1. Prototype: https://bahdigital.invisionapp.com/share/EYIADACT3Z8
2. Content: [Caution Flag Copy for Review v.2.0.xlsx](https://github.com/department-of-veterans-affairs/va.gov-team/files/4376587/Caution.Flag.Copy.for.Review.v.2.0.xlsx)

**Comment by zurbergram** [comment]
Caution flag and reason are set at
- https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L172
- https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L263
- https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L352
- https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L378
- https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L398

Used at
- https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/gi/components/profile/CautionaryInformation.jsx#L125

Tl;dr; of logic for how current fields are setup and used
<img width="1648" alt="Screen Shot 2020-03-13 at 10 53 22 AM" src="https://user-images.githubusercontent.com/1094999/76632464-a977a600-6519-11ea-97c5-328d69a08705.png">
[Institutions_profile.xlsx](https://github.com/department-of-veterans-affairs/va.gov-team/files/4330158/Institutions_profile.xlsx)

**Comment by dmccurdybah** [comment]
Can be tested with the following branches:

vets-website: `gi-caution-flags`

gibct-data-service: `gi-caution-flags`

**Comment by sonntag-adam** [comment]
-- Set caution flag fields
Update caution_flags
SET
	title = 'Test caution flag',
	description = 'Test caution flag description',
	link_text = 'Click this link',
	link_url = 'https://va.gov'
WHERE id = 1234
-- institution with future closing
UPDATE institutions
SET
	school_closing = true,
	school_closing_on = '2022-1-1'
WHERE facility_code = '31000040'
-- institution closed
UPDATE institutions
SET school_closing = true
WHERE facility_code = '31000040'

**Comment by sonntag-adam** [comment]
-- Find caution flags for an institution:
SELECT caution_flags.*
FROM caution_flags
	INNER JOIN institutions ON institutions.id = caution_flags.institution_id
WHERE facility_code = '21014523'

**Comment by delli-gatti-michael** [comment]
PR:
https://github.com/department-of-veterans-affairs/vets-website/pull/12003

**Comment by delli-gatti-michael** [comment]
Local testing complete.

Screenshots with test data:
![Screen Shot 2020-03-31 at 11.36.42 AM.png](https://images.zenhubusercontent.com/5c07f000391a6f6357adc430/4d5aa7cc-296f-4bee-b758-4ddf80734e2a)
![Screen Shot 2020-03-31 at 9.24.08 AM.png](https://images.zenhubusercontent.com/5c07f000391a6f6357adc430/8ce13bb3-f022-40e0-8077-650af02faf93)

**Comment by sonntag-adam** [comment]
https://dsvavsp.testrail.io/index.php?/runs/view/243&group_by=cases:section_id&group_order=asc

**Comment by sonntag-adam** [comment]
Closing story, fixes will be completed in https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7743
